internal/ledgerutil/compare.go

```diff
@@ -39,8 +44,20 @@ func Compare(snapshotDir1 string, snapshotDir2 string, outputFile string) (count
 		return 0, errors.New("both snapshots public state hashes are same. Aborting compare")
 	}
 
+	// Output directory creation
+	outputDirName := filepath.Base(snapshotDir1) + "_and_" + filepath.Base(snapshotDir2) + "_compared"
```

Comment:
In my trial, I used snapshots of block height 4. This resulted in a directory of the name "4_and_4_compared". The block height will typically be the same for comparisons, so including it in the directory name doesn't make sense to me. Additionally, I think Manish mentioned earlier that we don't need to create new directories, we can just put the results directly in the output directory that the user specified.

Reply:
The reason I decided to add back code for creating a new directory was because eventually we are going to be outputting more files like the early divergences results and the private hash comparison results. If we want to give the responsibility of creating and naming the directory to the user, wouldn't it make more sense to create the directory within the tool by allowing the user to pass in the name they want to use? We could also have a generic name for cases where the user doesn't pass anything in. This would allow us to keep the same current functionality and eliminate confusing output directory names.

Reply:
Yes, we will be outputting multiple files, but all those files can go into the single output directory that the user specified (or in the current directory if they did not specify an output directory).  I'm still need seeing a need to create a subdirectory.

Reply:
If we go this route, should we also follow Manish's suggestion and only allow the command to successfully generate output files if the user specified directory isn't empty? If so, this was the other purpose of creating an output directory, to ensure the output directory would be empty as to not overwrite existing output files. Otherwise, we would need to check for the existence of each output file individually, which Manish was against.